framework/QuartzRenderer/QuartzRenderer.swift

```diff
@@ -0,0 +1,599 @@
+                                     green: CGFloat(fillColor.g),
+                                     blue: CGFloat(fillColor.b),
+                                     alpha: CGFloat(fillColor.a)))
+        context.addPath(polygonPath)
+        context.fillPath()
+    }
+
+    public func drawLine(startPoint p1: Point,
+                         endPoint p2: Point,
+                         strokeWidth thickness: Float,
+                         strokeColor: Color = Color.black,
+                         isDashed: Bool,
+                         isOriginShifted: Bool) {
+        let line = CGMutablePath()
+        if (isOriginShifted) {
+            line.move(to: CGPoint(x: Double(p1.x + 0.1*plotDimensions.subWidth + xOffset),
+                                  y: Double(p1.y + 0.1*plotDimensions.subHeight + yOffset)))
+            line.addLine(to: CGPoint(x: Double(p2.x + 0.1*plotDimensions.subWidth + xOffset),
+                                     y: Double(p2.y + 0.1*plotDimensions.subHeight + yOffset)))
+        }
+        else {
+            line.move(to: CGPoint(x: Double(p1.x + xOffset), y: Double(p1.y + yOffset)))
+            line.addLine(to: CGPoint(x: Double(p2.x + xOffset), y: Double(p2.y + yOffset)))
+        }
+        context.setStrokeColor(CGColor(red: CGFloat(strokeColor.r),
+                                       green: CGFloat(strokeColor.g),
+                                       blue: CGFloat(strokeColor.b),
+                                       alpha: CGFloat(strokeColor.a)))
+        context.setLineWidth(CGFloat(thickness))
+        context.addPath(line)
+        if(isDashed) {
+            let dashes: [ CGFloat ] = [ CGFloat(thickness + 1), CGFloat(thickness + 1) ]
+            context.setLineDash(phase: 1, lengths: dashes)
+        }
+        context.strokePath()
+        context.setLineDash(phase: 1, lengths: [])
+    }
+
+    public func drawPlotLines(points p: [Point],
+                              strokeWidth thickness: Float,
+                              strokeColor: Color,
+                              isDashed: Bool) {
+        for i in 0..<p.count-1 {
+            drawLine(startPoint: p[i],
+                     endPoint: p[i+1],
+                     strokeWidth: thickness,
+                     strokeColor: strokeColor,
+                     isDashed: isDashed,
+                     isOriginShifted: true)
+        }
+    }
+
+    public func drawText(text s: String,
+                         location p: Point,
+                         textSize size: Float,
+                         strokeWidth thickness: Float,
+                         angle: Float,
+                         isOriginShifted: Bool){
+        var x1 = p.x + xOffset
+        var y1 = p.y + yOffset
+        if (isOriginShifted) {
+            x1 = x1 + 0.1*plotDimensions.subWidth
+            y1 = y1 + 0.1*plotDimensions.subHeight
+        }
+        let font = NSFont.systemFont(ofSize: CGFloat(size))
```

Comment:
NSFont is a another platform-specific type, so you might need to wrap this in an `#if os(macOS` and have a matching path using UIFont for iOS.